### PR TITLE
Change argument order of JIT Helpers to proper C linkage order

### PR DESCRIPTION
**Pull request by 0dvictor**
Argument order of VM's JIT Helpers were the order of Java private linkage,
and the order was kept for historical reasons. Now it is the time to be
corrected.

Signed-off-by: Victor Ding <dvictor@ca.ibm.com>

### CODE REVIEW

**Comment by r30shah** [comment]
@0dvictor Thanks for making changes. Will have a crack at this change soon.

File: runtime/oti/jitprotos.h [hunk 1]
```diff
@@ -173,7 +173,12 @@ void J9FASTCALL fast_jitVolatileWriteDouble(J9VMThread *currentThread, U_64 *add
 #endif /* !J9VM_ENV_DATA64 */
 void J9FASTCALL fast_jitCheckIfFinalizeObject(J9VMThread *currentThread, j9object_t object);
 void J9FASTCALL fast_jitCollapseJNIReferenceFrame(J9VMThread *currentThread);
+#ifdef J9VM_ARCH_X86
+// TODO Will be cleaned once all platforms adopt the correct parameter order
```
Comment (gacholio) [inline runtime/oti/jitprotos.h]: This is definitely not allowed - no // comments in .h files.

**Comment by andrewcraik** [comment]
Jenkins test sanity

**Comment by 0dvictor** [comment]
Jenkins test sanity

**Pull request merged**